docs: add known issue cross-arch building

cdrage opened the pull request:
docs: add known issue cross-arch building

### What does this PR do?

* Adds documentation regarding cross-arch building
* Docs are meant as an intermediary until podman machine is patched with
  the new QEMU calls
* When QEMU is patched, the only thing needed is to "disable rosetta"

### Screenshot / video of UI

<!-- If this PR is changing UI, please include
screenshots or screencasts showing the difference -->

N/A

### What issues does this PR fix or reference?

<!-- Include any related issues from Podman Desktop
repository (or from another issue tracker). -->

References https://github.com/containers/podman-desktop-extension-bootc/issues/808

### How to test this PR?

<!-- Please explain steps to reproduce -->

Follow documentation instructions on how to cross-arch build again.

Signed-off-by: Charlie Drage <charlie@charliedrage.com>

deboer-tim reviewed (commented):
First 5 lines, linking to upstream bug - excellent.

Next bit - I'm not sure we should be explaining the cause of the bug here or how it will be resolved, would prefer to see that upstream.

'follow these steps to resolve' followed by 'preliminary steps' - It's not clear to me as a Mac user whether this will fix the problem or these are steps I will need to take before the issue is resolved. If the latter, once the fix comes will I still need to take these steps? (and again, feels like this should be upstream)

As a Windows user, I'm assuming this issue applies to me? But 'follow these steps to resolve' only list Mac, would be nice to confirm platforms and that we don't have steps for others at the moment.

cdrage commented:
> First 5 lines, linking to upstream bug - excellent.
> 
> Next bit - I'm not sure we should be explaining the cause of the bug here or how it will be resolved, would prefer to see that upstream.

Upstream fix has already been applied for QEMU.

However, Rosetta must be disabled even after the QEMU fix goes in, so we'll still have to have this section :(. We are a special edge case that uses these openat2 system calls which are causing issues in the macOS kernel with Rosetta. 

> 
> 'follow these steps to resolve' followed by 'preliminary steps' - It's not clear to me as a Mac user whether this will fix the problem or these are steps I will need to take before the issue is resolved. If the latter, once the fix comes will I still need to take these steps? (and again, feels like this should be upstream)

Updated the headers to indicate more clearly it's for macOS and Windows.

> 
> As a Windows user, I'm assuming this issue applies to me? But 'follow these steps to resolve' only list Mac, would be nice to confirm platforms and that we don't have steps for others at the moment.

Updated the steps! :)

jeffmaury reviewed (commented):
I'm wondering if we can't use the RPM repo URL instead of direct link to the RPM that will ensure if a new build is created we will get it

cdrage commented:
> I'm wondering if we can't use the RPM repo URL instead of direct link to the RPM that will ensure if a new build is created we will get it

No new builds will be created since it's already been merged into QEMU upstream, so just waiting for podman to update their machine images / qemu to do a new release as well.
